### ISSUE TITLE
Detected another IPv4 mDNS stack running on this host

### ISSUE THREAD

**Issue by eximo84** [linuxserver/docker-daapd#63]
[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

Server rebooted due to power cut, when server came back up and docker containers started i noticed forked-daapd not working.  Looking at the logs i can see the service browser isnt working so my devices are not listed and my remote app isnt connecting or accepting new connections.

------------------------------

## Expected Behavior

AirPlay devices showing in the UI,remote app connecting

## Current Behavior

Nothing displaying in outputs other than localhost, remote isnt showing pairing.

## Steps to Reproduce

1.
2.
3.
4.

## Environment
**OS:** Debian
**CPU architecture:** x86_64
**How docker service was installed:** official docker

## Command used to create docker container (run/create/compose/screenshot)

version: "2.1"
services:
  owntone:
    image: lscr.io/linuxserver/daapd:latest
    container_name: owntone
    network_mode: host
    environment:
      - PUID=1002
      - PGID=100
      - TZ=Europe/London
    volumes:
      - /srv/dev-disk-by-label-data/Home/dockeruser/.config/owntone:/config
      - /srv/dev-disk-by-label-PrimaryRaid/Media/Music:/music
    restart: unless-stopped

## Docker logs

[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-envfile: executing... 
[cont-init.d] 01-envfile: exited 0.
[cont-init.d] 01-migrations: executing... 
[migrations] started
[migrations] no migrations found
[cont-init.d] 01-migrations: exited 0.
[cont-init.d] 02-tamper-check: executing... 
[cont-init.d] 02-tamper-check: exited 0.
[cont-init.d] 10-adduser: executing... 

-------------------------------------
          _         ()
         | |  ___   _    __
         | | / __| | |  /  \
         | | \__ \ | | | () |
         |_| |___/ |_|  \__/

Brought to you by linuxserver.io
-------------------------------------

To support LSIO projects visit:
https://www.linuxserver.io/donate/
-------------------------------------
GID/UID
-------------------------------------

User uid:    1002
User gid:    100
-------------------------------------

[cont-init.d] 10-adduser: exited 0.
[cont-init.d] 30-dbus: executing... 
[cont-init.d] 30-dbus: exited 0.
[cont-init.d] 40-config: executing... 
[cont-init.d] 40-config: exited 0.
[cont-init.d] 90-custom-folders: executing... 
[cont-init.d] 90-custom-folders: exited 0.
[cont-init.d] 99-custom-files: executing... 
[custom-init] no custom files found exiting...
[cont-init.d] 99-custom-files: exited 0.
[cont-init.d] done.
[services.d] starting services
[2022-05-09T21:14:33Z INFO  librespot] librespot 0.3.1 866c55ab88 (Built on 2021-11-03, Build ID: LYkZUkEY, Profile: release)
[2022-05-09T21:14:33Z WARN  libmdns] Failed to register IPv6 receiver: Os { code: 19, kind: Uncategorized, message: "No such device" }
[services.d] done.
Failed to set ownership on logfile: Operation not permitted
[2022-05-09 22:14:33] [  LOG]     main: OwnTone version 28.3 taking off
[2022-05-09 22:14:33] [  LOG]     main: Built with:
[2022-05-09 22:14:33] [  LOG]     main: - ffmpeg
[2022-05-09 22:14:33] [  LOG]     main: - Spotify
[2022-05-09 22:14:33] [  LOG]     main: - librespot-c
[2022-05-09 22:14:33] [  LOG]     main: - LastFM
[2022-05-09 22:14:33] [  LOG]     main: - Chromecast
[2022-05-09 22:14:33] [  LOG]     main: - MPD
[2022-05-09 22:14:33] [  LOG]     main: - Websockets
[2022-05-09 22:14:33] [  LOG]     main: - ALSA
[2022-05-09 22:14:33] [  LOG]     main: - Webinterface
[2022-05-09 22:14:33] [  LOG]     main: - Regex
[2022-05-09 22:14:33] [  LOG]     main: mDNS init
[2022-05-09 22:14:33] [  LOG]     mdns: Avahi state change: Client connecting
[2022-05-09 22:14:33] [  LOG]       db: Configured to use database file '/config/dbase_and_logs/songs3.db'
[2022-05-09 22:14:33] [  LOG]       db: Now vacuuming database, this may take some time...
Found user 'avahi' (UID 86) and group 'avahi' (GID 86).
Successfully dropped root privileges.
avahi-daemon 0.8 starting up.
WARNING: No NSS support for mDNS detected, consider installing nss-mdns!
Loading service file /etc/avahi/services/sftp-ssh.service.
Loading service file /etc/avahi/services/ssh.service.
*** WARNING: Detected another IPv4 mDNS stack running on this host. This makes mDNS unreliable and is thus not recommended. ***
Joining mDNS multicast group on interface br-e234c8232193.IPv4 with address 172.29.0.1.
New relevant interface br-e234c8232193.IPv4 for mDNS.
Joining mDNS multicast group on interface br-a16bec8428c6.IPv4 with address 172.30.0.1.
New relevant interface br-a16bec8428c6.IPv4 for mDNS.
Joining mDNS multicast group on interface br-702bc3e33157.IPv4 with address 192.168.128.1.
New relevant interface br-702bc3e33157.IPv4 for mDNS.
Joining mDNS multicast group on interface br-6053c4365e7d.IPv4 with address 192.168.112.1.
New relevant interface br-6053c4365e7d.IPv4 for mDNS.
Joining mDNS multicast group on interface br-1f282f4fc32b.IPv4 with address 172.25.0.1.
New relevant interface br-1f282f4fc32b.IPv4 for mDNS.
Joining mDNS multicast group on interface br-0b01510a5bb6.IPv4 with address 172.20.0.1.
New relevant interface br-0b01510a5bb6.IPv4 for mDNS.
Joining mDNS multicast group on interface br-e64617624b5e.IPv4 with address 172.19.0.1.
New relevant interface br-e64617624b5e.IPv4 for mDNS.
Joining mDNS multicast group on interface br-6cf1a195c7f5.IPv4 with address 172.22.0.1.
New relevant interface br-6cf1a195c7f5.IPv4 for mDNS.
Joining mDNS multicast group on interface br-4bcdeb88ce8c.IPv4 with address 192.168.80.1.
New relevant interface br-4bcdeb88ce8c.IPv4 for mDNS.
Joining mDNS multicast group on interface br-292fc9c0ec2e.IPv4 with address 172.18.0.1.
New relevant interface br-292fc9c0ec2e.IPv4 for mDNS.
Joining mDNS multicast group on interface br-e48d15e1becd.IPv4 with address 172.26.0.1.
New relevant interface br-e48d15e1becd.IPv4 for mDNS.
Joining mDNS multicast group on interface br-c84a1bcf3614.IPv4 with address 172.21.0.1.
New relevant interface br-c84a1bcf3614.IPv4 for mDNS.
Joining mDNS multicast group on interface br-a3cc031e6b55.IPv4 with address 172.23.0.1.
New relevant interface br-a3cc031e6b55.IPv4 for mDNS.
Joining mDNS multicast group on interface br-12f68511d69c.IPv4 with address 192.168.48.1.
New relevant interface br-12f68511d69c.IPv4 for mDNS.
Joining mDNS multicast group on interface br-08af05386568.IPv4 with address 192.168.16.1.
New relevant interface br-08af05386568.IPv4 for mDNS.
Joining mDNS multicast group on interface docker0.IPv4 with address 172.17.0.1.
New relevant interface docker0.IPv4 for mDNS.
Joining mDNS multicast group on interface br-bf67c9e412bf.IPv4 with address 192.168.32.1.
New relevant interface br-bf67c9e412bf.IPv4 for mDNS.
Joining mDNS multicast group on interface br-bce2c5703cd7.IPv4 with address 192.168.96.1.
New relevant interface br-bce2c5703cd7.IPv4 for mDNS.
Joining mDNS multicast group on interface br-91a65f2cc3ec.IPv4 with address 172.24.0.1.
New relevant interface br-91a65f2cc3ec.IPv4 for mDNS.
Joining mDNS multicast group on interface br-4bc2355ae7e5.IPv4 with address 192.168.144.1.
New relevant interface br-4bc2355ae7e5.IPv4 for mDNS.
Joining mDNS multicast group on interface enp1s0.IPv4 with address 192.168.1.10.
IP_ADD_MEMBERSHIP failed: No buffer space available
Joining mDNS multicast group on interface lo.IPv6 with address ::1.
New relevant interface lo.IPv6 for mDNS.
Joining mDNS multicast group on interface lo.IPv4 with address 127.0.0.1.
IP_ADD_MEMBERSHIP failed: No buffer space available
Network interface enumeration completed.
Registering new address record for 172.29.0.1 on br-e234c8232193.IPv4.
Registering new address record for 172.30.0.1 on br-a16bec8428c6.IPv4.
Registering new address record for 192.168.128.1 on br-702bc3e33157.IPv4.
Registering new address record for 192.168.112.1 on br-6053c4365e7d.IPv4.
Registering new address record for 172.25.0.1 on br-1f282f4fc32b.IPv4.
Registering new address record for 172.20.0.1 on br-0b01510a5bb6.IPv4.
Registering new address record for 172.19.0.1 on br-e64617624b5e.IPv4.
Registering new address record for 172.22.0.1 on br-6cf1a195c7f5.IPv4.
Registering new address record for 192.168.80.1 on br-4bcdeb88ce8c.IPv4.
Registering new address record for 172.18.0.1 on br-292fc9c0ec2e.IPv4.
Registering new address record for 172.26.0.1 on br-e48d15e1becd.IPv4.
Registering new address record for 172.21.0.1 on br-c84a1bcf3614.IPv4.
Registering new address record for 172.23.0.1 on br-a3cc031e6b55.IPv4.
Registering new address record for 192.168.48.1 on br-12f68511d69c.IPv4.
Registering new address record for 192.168.16.1 on br-08af05386568.IPv4.
Registering new address record for 172.17.0.1 on docker0.IPv4.
Registering new address record for 192.168.32.1 on br-bf67c9e412bf.IPv4.
Registering new address record for 192.168.96.1 on br-bce2c5703cd7.IPv4.
Registering new address record for 172.24.0.1 on br-91a65f2cc3ec.IPv4.
Registering new address record for 192.168.144.1 on br-4bc2355ae7e5.IPv4.
Registering new address record for 192.168.1.10 on enp1s0.IPv4.
Registering new address record for ::1 on lo.*.
Registering new address record for 127.0.0.1 on lo.IPv4.
[2022-05-09 22:14:34] [  LOG]       db: Database OK with 6964 active files and 6 active playlists
[2022-05-09 22:14:34] [  LOG]     mdns: Failed to create service browser: Bad state
[2022-05-09 22:14:34] [  LOG]     raop: Could not add mDNS browser for AirPlay devices
[2022-05-09 22:14:34] [  LOG]     mdns: Failed to create service browser: Bad state
[2022-05-09 22:14:34] [  LOG]  airplay: Could not add mDNS browser for AirPlay devices
ALSA lib confmisc.c:165:(snd_config_get_card) Cannot get card index for 0
[2022-05-09 22:14:34] [  LOG]     mdns: Failed to create service browser: Bad state
[2022-05-09 22:14:34] [  LOG]     cast: Could not add mDNS browser for Chromecast devices
[2022-05-09 22:14:35] [  LOG]     mdns: Failed to create service browser: Bad state
[2022-05-09 22:14:35] [FATAL]   remote: Could not browse for Remote services
[2022-05-09 22:14:35] [FATAL]     main: Remote pairing service failed to start
[2022-05-09 22:14:35] [  LOG]     main: MPD deinit
[2022-05-09 22:14:35] [  LOG]     main: HTTPd deinit
[2022-05-09 22:14:35] [  LOG]     main: Player deinit
[2022-05-09 22:14:35] [  LOG]     scan: Scanned 200 files...
Server startup complete. Host name is homenas.local. Local service cookie is 440781369.
[2022-05-09 22:14:35] [  LOG]     scan: Scanned 400 files...
[2022-05-09 22:14:35] [  LOG]     main: Library scanner deinit
[2022-05-09 22:14:35] [  LOG]  spotify: No spotify refresh token found
[2022-05-09 22:14:35] [  LOG]  spotify: Spotify webapi token refresh failed. In order to use Spotify, authorize the server to access your saved tracks by visiting http://owntone.local:3689
Service "homenas" (/etc/avahi/services/ssh.service) successfully established.
Service "homenas" (/etc/avahi/services/sftp-ssh.service) successfully established.
[2022-05-09 22:14:38] [  LOG]      lib: Library init scan completed in 4 sec (3 changes)
Failed to set ownership on logfile: Operation not permitted
[2022-05-09 22:14:38] [  LOG]     main: OwnTone version 28.3 taking off
[2022-05-09 22:14:38] [  LOG]     main: Built with:
[2022-05-09 22:14:38] [  LOG]     main: - ffmpeg
[2022-05-09 22:14:38] [  LOG]     main: - Spotify
[2022-05-09 22:14:38] [  LOG]     main: - librespot-c
[2022-05-09 22:14:38] [  LOG]     main: - LastFM
[2022-05-09 22:14:38] [  LOG]     main: - Chromecast
[2022-05-09 22:14:38] [  LOG]     main: - MPD
[2022-05-09 22:14:38] [  LOG]     main: - Websockets
[2022-05-09 22:14:38] [  LOG]     main: - ALSA
[2022-05-09 22:14:38] [  LOG]     main: - Webinterface
[2022-05-09 22:14:38] [  LOG]     main: - Regex
[2022-05-09 22:14:38] [  LOG]     main: mDNS init
[2022-05-09 22:14:38] [  LOG]     mdns: Avahi state change: Client running
[2022-05-09 22:14:38] [  LOG]       db: Configured to use database file '/config/dbase_and_logs/songs3.db'
[2022-05-09 22:14:38] [  LOG]       db: Now vacuuming database, this may take some time...
[2022-05-09 22:14:38] [  LOG]       db: Database OK with 385 active files and 6 active playlists
ALSA lib confmisc.c:165:(snd_config_get_card) Cannot get card index for 0
[2022-05-09 22:14:38] [  LOG]     scan: Scanned 200 files...
[2022-05-09 22:14:39] [  LOG]     scan: Scanned 400 files...

**Comment by eximo84** [comment]
well after two evenings and lots of duck duck going I put this on a raspberry pi with docker and it worked which proved it was something in my system. Went through logs comparing mine to the Rpi and saw similar issues with mdns and bad state however it was working ok. 

I fixed one error to do with dbus by adding device option to my compose file then I stumbled across a post where someone ran a docker network prune. Ran this expecting nothing, recreated my container and it worked. 

Still got random errors but I can see my speakers and connect via my phone.
